libshaderc/include/shaderc.hpp

```diff
@@ -125,6 +125,14 @@ class CompileOptions {
     shaderc_compile_options_set_generate_debug_info(options_);
   }
 
+  void SetIncluderCallbacks(
```

Comment:
If we are exposing a C++ api, would it make sense to not use raw C functions and either
A) have this take an object that implements an INCLUDER interface or
```
void SetIncluder(std::unique_ptr<Includer>&& includer)
```
or at least

B) void SetIncluder(std::function<...> get_includer_response,
                                std::function<...> release_includer_response)
// Note there is no user_data* since std::function can contain a
// struct/function/lambda.